src/Symfony/Component/Config/Definition/Builder/TreeBuilder.php

```diff
@@ -23,6 +23,15 @@ class TreeBuilder implements NodeParentInterface
     protected $tree;
     protected $root;
 
+    public function __construct(string $name = null, string $type = 'array', NodeBuilder $builder = null)
```

Comment:
how about `?NodeBuilder` instead ? Use a nullable typehint instead of relying on 7.0's hackish way.

Reply:
There is nothing hackish here, optional arguments are just fit here IMHO.

Reply:
@Taluu it's not "more pertinent", is *exactly* the same, meaning wise. It's not a hack at all but how the language is defined.

Reply:
```php
<?php

interface A {
    public function dummy(int $arg = null);
}

class B implements A {
    public function dummy(int $arg) { }
}
```

> Fatal error: Declaration of B::dummy(int $arg) must be compatible with A::dummy(?int $arg = NULL) in nullable.php on line 7

Note the `?` in the stack trace while it is not in the signature, it's really the same for the engine